packages/api/app/Controllers/Http/InstitutionController.js

```diff
@@ -2,17 +2,34 @@
 const Institution = use('App/Models/Institution');
 const Upload = use('App/Models/Upload');
 const User = use('App/Models/User');
-const { errors, errorPayload, getTransaction, Algolia } = require('../../Utils');
+const {
+	errors,
+	errorPayload,
+	getTransaction,
+	Algolia,
+	technologyStatuses,
+} = require('../../Utils');
 
 const saveAlgoliaIndex = async (institutionId) => {
 	const institution = await Institution.query()
 		.where({ id: institutionId })
 		.with('logo')
-		.withCount('technologies')
-		.withCount('services')
-		.fetch();
-
-	await Algolia.saveIndex('institution', institution);
+		.withCount('technologies', (builder) => {
+			builder
+				.where('technologies.status', technologyStatuses.PUBLISHED)
+				.where('active', true);
+		})
+		.withCount('services', (builder) => {
+			builder.where('active', true);
+		})
+		.first();
+
+	if (
+		institution.$sideLoaded.technologies_count > 0 ||
+		institution.$sideLoaded.services_count > 0
+	) {
+		await Algolia.saveIndex('institution', institution);
+	}
```

Comment:
We don't need it here. Otherwise, we would have to delete or include items in Algolia after create/delete a service. It's better to do as it is now: send **everyone** to Algolia and make this filter in the frontend

```suggestion
	await Algolia.saveIndex('institution', institution);
```